pkg/booleanpolicy/default_policies_test.go

```diff
@@ -2272,6 +2276,82 @@ func (suite *DefaultPoliciesTestSuite) TestImageVerified() {
 	}
 }
 
+func (suite *DefaultPoliciesTestSuite) TestImageVerified_WithDeployment() {
+	const (
+		verifier1 = "io.stackrox.signatureintegration.00000000-0000-0000-0000-000000000002"
+		verifier2 = "io.stackrox.signatureintegration.00000000-0000-0000-0000-000000000003"
+		verifier3 = "io.stackrox.signatureintegration.00000000-0000-0000-0000-000000000004"
+	)
+
+	images := []*storage.Image{
+		imageWithSignatureVerificationResults("image_verified_by_1", []*storage.ImageSignatureVerificationResult{
+			{
+				VerifierId:              verifier1,
+				Status:                  storage.ImageSignatureVerificationResult_VERIFIED,
+				VerifiedImageReferences: []string{"image_verified_by_1"},
+			},
+		}),
+		imageWithSignatureVerificationResults("image_with_alternative_verified_reference", []*storage.ImageSignatureVerificationResult{
+			{
+				VerifierId:              verifier2,
+				Status:                  storage.ImageSignatureVerificationResult_VERIFIED,
+				VerifiedImageReferences: []string{"image_verified_by_2"},
+			},
+		}),
+		imageWithSignatureVerificationResults("image_verified_by_2", []*storage.ImageSignatureVerificationResult{
+			{
+				VerifierId:              verifier3,
+				Status:                  storage.ImageSignatureVerificationResult_VERIFIED,
+				VerifiedImageReferences: []string{"image_with_alternative_verified_reference", "image_verified_by_2"},
+			},
+		}),
+	}
+
+	cases := map[string]struct {
+		deployment       *storage.Deployment
+		image            *storage.Image
+		matchingVerifier string
+		expectViolation  bool
+	}{
+		"deployment with matching verified image reference shouldn't lead in alert message": {
+			deployment:       deploymentWithImage("deployment_with_image_verified_by_1", images[0]),
+			image:            images[0],
```

Comment:
nit: let's extract these to variables with names like `imageWithMatchingResults` etc to improve readability

Reply:
I've extracted those to variable names, let's hope it's a bit more clear and readable.